libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -737,7 +737,38 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		if (arguments.empty())
 			m_code << "revert(0, 0)\n";
 		else
-			solUnimplementedAssert(false, "");
+		{
+			solAssert(arguments.size() == 1, "");
+			solAssert(arguments.front()->annotation().type->isImplicitlyConvertibleTo(
+				*TypeProvider::fromElementaryTypeName("string memory")),
+				""
+			);
+
+			Whiskers templ(R"({
+				let <pos> := <allocateTemporary>()
+				mstore(<pos>, <hash>)
+				let <end> := <encode>(add(<pos>, 4) <argumentVars>)
+				revert(<pos>, sub(<end>, <pos>))
+			})");
+			templ("pos", m_context.newYulVariable());
+			templ("end", m_context.newYulVariable());
+			templ(
+				"hash",
+				(u256(util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)")))) << (256 - 32)).str()
+			);
+			templ("allocateTemporary", m_utils.allocationTemporaryMemoryFunction());
+			templ(
+				"argumentVars",
+				(arguments.front()->annotation().type->sizeOnStack() > 0 ? ", " : "") +
+					suffixedVariableNameList("message_", 1, 1 + arguments.front()->annotation().type->sizeOnStack())
```

Comment:
I'm not so sure if it is a good idea to add these prefixes - it could lead to collisions.

Also, the arguments might come from local variables, then they have their own name already anyway, don't they?

Reply:
Yes, this was very bad idea, will update now